src/Datadog.Trace.ClrProfiler.Managed/Integrations/ElasticsearchNetCommon.cs

```diff
@@ -47,19 +45,24 @@ public static Scope CreateScope(Tracer tracer, string integrationName, object pi
 
             try
             {
-                scope = tracer.StartActive(OperationName, serviceName: serviceName);
+                var tags = new ElasticsearchTags();
+                scope = tracer.StartActiveWithTags(OperationName, serviceName: serviceName, tags: tags);
                 var span = scope.Span;
                 span.ResourceName = requestName ?? pathAndQuery ?? string.Empty;
                 span.Type = SpanType;
-                span.SetTag(Tags.InstrumentationName, ComponentValue);
-                span.SetTag(Tags.SpanKind, SpanKinds.Client);
-                span.SetTag(ElasticsearchActionKey, requestName);
-                span.SetTag(ElasticsearchMethodKey, method);
-                span.SetTag(ElasticsearchUrlKey, url);
+                tags.InstrumentationName = ComponentValue;
+                tags.SpanKind = SpanKinds.Client;
+                tags.Action = requestName;
+                tags.Method = method;
+                tags.Url = url;
 
                 // set analytics sample rate if enabled
                 var analyticsSampleRate = tracer.Settings.GetIntegrationAnalyticsSampleRate(integrationName, enabledWithGlobalSetting: false);
-                span.SetMetric(Tags.Analytics, analyticsSampleRate);
+
+                if (analyticsSampleRate != null)
+                {
+                    tags.AnalyticsSampleRate = analyticsSampleRate;
+                }
```

Comment:
There's a lot of integrations with this block, can we just refactor to a new `SampledSpanCommonTags : CommonTags` with this kind of logic there, or something to avoid repeat the same lines on severals integrations?

Reply:
All integrations should have this. The only difference is that some (e.g. web frameworks) should set `enabledWithGlobalSetting: true` and everything else should set `enabledWithGlobalSetting: false`.

Reply:
Refactoring this would be better if all automated spans had the InstrumentationName tag, and I feel like they should (just like SpanKind). I'll do that in the same PR as spankind